rust/stackable-cockpit/src/platform/cluster.rs

```diff
@@ -77,3 +83,65 @@ impl ClusterInfo {
         })
     }
 }
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct ResourceRequests {
+    pub cpu: Quantity,
+    pub memory: Quantity,
+    pub pvc: Quantity,
+}
+
+impl Display for ResourceRequests {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "CPU: {}, Memory: {}, PVC space: {}",
+            self.cpu.0, self.memory.0, self.pvc.0
+        )
+    }
+}
+
+/// This error indicates that the ResourceRequirements of a stack or a demo
+/// can not be parsed
+#[derive(Debug, Snafu)]
+pub enum ResourceRequestsError {
+    #[snafu(display("kube error: {source}"), context(false))]
+    KubeError { source: KubeClientError },
+    #[snafu(display("failed to parse cpu resource requirements"))]
+    ParseCpuResourceRequirements {
+        source: stackable_operator::error::Error,
+    },
+    #[snafu(display("failed to parse memory resource requirements"))]
+    ParseMemoryResourceRequirements {
+        source: stackable_operator::error::Error,
+    },
+}
+
+impl ResourceRequests {
+    /// `object_name` should be `Stack` or `Demo`.
+    pub async fn warn_when_cluster_too_small(
+        &self,
+        object_name: &str,
+    ) -> Result<(), ResourceRequestsError> {
+        let kube_client = KubeClient::new().await?;
+        let cluster_info = kube_client.get_cluster_info().await?;
+
+        let stack_cpu =
+            CpuQuantity::try_from(&self.cpu).context(ParseCpuResourceRequirementsSnafu)?;
+        let stack_memory =
+            MemoryQuantity::try_from(&self.memory).context(ParseMemoryResourceRequirementsSnafu)?;
+
+        if stack_cpu > cluster_info.untainted_allocatable_cpu {
+            warn!(
+                "{object_name} has resource requirements [{self}], but cluster seems to have only {} cores",
```

Comment:
I did prefer to print out the whole stack resource requirements, so that you don't see the error message: "To few CPU", add some cores just to be surprised that you also have to few Memory. I think this could be prevented by directly printing all requirements but no strong opinion.